Makefile

```diff
@@ -23,3 +23,17 @@ test: fmt lint
 	CARGO_GENERATE_VALUE_REGISTRY=ghcr.io \
 	CARGO_GENERATE_VALUE_REGISTRY_MODULE_PATH_PREFIX='kubewarden/policies' \
 	cargo generate --test
+
+TMPDIR:=$(shell mktemp -d --dry-run)
+
+e2e-tests:
+	# We cannot use cargo-generate --test for e2e-tests, since we need to access
+	# the project directory to access the e2e tests and the annotated wasm module.
+	CARGO_GENERATE_VALUE_PROJECT_NAME=demo \
+	CARGO_GENERATE_VALUE_REGISTRY=ghcr.io \
+	CARGO_GENERATE_VALUE_REGISTRY_MODULE_PATH_PREFIX='kubewarden/policies' \
+	cargo generate --path . --destination $(TMPDIR)
+	# Note about the leading `-` symbol: we ignore failures of the test suite to
+	# ensure the cleanup code always run
+	-cd $(TMPDIR)/demo && make e2e-tests
```

Comment:
shouldn't we fail if the e2e tests fail? if not the GHA will never fail.
I'm ok with leaving the tmpdir around as it is a tmpdir, or cleaning it in a different way.

Reply:
Good point, I've changed the code to avoid this behavior